File: app/src/main/java/com/gnoemes/shikimoriapp/domain/anime/series/AlternativeSeriesInteractor.java
package com.gnoemes.shikimoriapp.domain.anime.series;

import com.gnoemes.shikimoriapp.entity.anime.series.domain.AlternativeTranslation;
import com.gnoemes.shikimoriapp.entity.anime.series.domain.AlternativeTranslationType;
import com.gnoemes.shikimoriapp.entity.anime.series.domain.Episode;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Single;

public interface AlternativeSeriesInteractor {

    /**
     * Get episodes
     */
    Single<List<Episode>> getEpisodes(long animeId);

    /**
     * Episode was Watched
     */
    Completable setEpisodeWatched(long animeId, long episodeId, long rateId);


    /**
     * Episode was Watched
     */
    Completable setEpisodeWatched(long animeId, long episodeId);

    /**
     * Get translations for episode
     */
    Single<List<AlternativeTranslation>> getEpisodeTranslations(AlternativeTranslationType type, long animeId, long episodeId);

    /**
     * Clear local history
     */
    Completable clearHistory(long animeId);
}

File: app/src/main/java/com/gnoemes/shikimoriapp/domain/anime/series/AlternativeSeriesInteractor.kt
package com.gnoemes.shikimoriapp.domain.anime.series

import com.gnoemes.shikimoriapp.entity.anime.series.domain.AlternativeTranslation
import com.gnoemes.shikimoriapp.entity.anime.series.domain.AlternativeTranslationType
import com.gnoemes.shikimoriapp.entity.anime.series.domain.Episode
import io.reactivex.Completable
import io.reactivex.Single

interface AlternativeSeriesInteractor {

    /**
     * Get episodes
     */
    fun getEpisodes(animeId: Long): Single<List<Episode>>

    /**
     * Episode was Watched
     */
    fun setEpisodeWatched(animeId: Long, episodeId: Long, rateId: Long = -1): Completable

    /**
     * Get translations for episode
     */
    fun getEpisodeTranslations(type: AlternativeTranslationType, animeId: Long, episodeId: Long): Single<List<AlternativeTranslation>>

    /**
     * Clear local history
     */
    fun clearHistory(animeId: Long): Completable

}